cfgov/v1/atomic_elements/organisms.py

```diff
@@ -360,8 +360,13 @@ def match_all_topic_tags(queryset, page_tags):
 
             related_items[parent.title()] = related_queryset[:limit]
 
-        # Return items in the dictionary that have non-empty querysets
-        return {key: value for key, value in related_items.items() if value}
+        # Return specific pages for each queryset in the dictionary that is not
+        # empty
+        return {
+            key: [v.specific for v in value]
```

Comment:
Is there some way of using [`.specific()`] on the querysets that are pulled back above, to reduce the number of database queries? As written this will do one DB lookup for each page that is returned in the queryset.

Reply:
I just completely overlooked `specific()` on the Wagtail `PageQueryset`! Changing to that now.

Reply:
Fixed in ef9cfa9!